extensions/wikia/AdEngine3/src/slots.js

```diff
@@ -0,0 +1,247 @@
+import { context, slotService, utils } from '@wikia/ad-engine';
+import { getAdProductInfo } from '@wikia/ad-engine/dist/ad-products';
+import { throttle } from 'lodash';
+
+const PAGE_TYPES = {
+  article: 'a',
+  home: 'h',
+};
+
+function setSlotState(slotName, state) {
+  if (state) {
+    slotService.enable(slotName);
+  } else {
+    slotService.disable(slotName);
+  }
+}
+
+function isIncontentBoxadApplicable() {
+  const isSupportedPageType = ['article', 'search'].indexOf(context.get('wiki.targeting.pageType')) !== -1;
+
+  return isSupportedPageType &&
+    window.wgIsContentNamespace &&
+    context.get('wiki.opts.adsInContent') &&
+    !context.get('wiki.targeting.wikiIsCorporate');
+}
+
+export default {
+  getContext() {
+    return {
+      TOP_LEADERBOARD: {
```

Comment:
Don't we wanna switch slot names to lowercase?

Reply:
Discussed offline. Not at this moment :)